libavfilter/qsvvpp.c

```diff
@@ -837,8 +841,17 @@ int ff_qsvvpp_filter_frame(QSVVPPContext *s, AVFilterLink *inlink, AVFrame *picr
                 ret = AVERROR(EAGAIN);
             break;
         }
-        out_frame->frame->pts = av_rescale_q(out_frame->surface.Data.TimeStamp,
-                                             default_tb, outlink->time_base);
+
+        if (ret == MFX_ERR_MORE_SURFACE &&
+            out_frame->surface.Data.TimeStamp == MFX_TIMESTAMP_UNKNOWN &&
+            s->last_in_pts != AV_NOPTS_VALUE)
+            dpts = (in_frame->frame->pts - s->last_in_pts) / 2;
```

Comment:
In frame conversion case, we may need 1 frame in N frame out, 
/ 2 may not right

Reply:
I didn't try FRC. It is possible the timestamp returned from the SDK is valid, otherwise FRC should be wrong even if without this patch. Anyway I will try FRC after cleanup the code.